### PR TITLE
Add the gnupg package and missing dependencies. Update versions of ex…

### PR DESCRIPTION
**Pull request by gartung**
…isting dependencies.

### CODE REVIEW

File: var/spack/repos/builtin/packages/gnupg/package.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+       standard as defined by RFC4880 """
+
+    homepage = "https://gnupg.org/index.html"
+    url      = "https://gnupg.org/ftp/gcrypt/gnupg/gnupg-2.1.21.tar.bz2"
+
+    version('2.1.21', '685ebf4c3a7134ba0209c96b18b2f064')
+
+    def configure_args(self):
+        args = []
+        return args
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gnupg/package.py]: Delete this function if not needed.

File: var/spack/repos/builtin/packages/libgcrypt/package.py [hunk 2]
```diff
@@ -32,8 +32,8 @@ class Libgcrypt(AutotoolsPackage):
        key algorithms, large integer functions, random numbers and a lot
        of supporting functions. """
     homepage = "http://www.gnu.org/software/libgcrypt/"
-    url      = "http://gd.tuwien.ac.at/pub/gnupg/libgcrypt/libgcrypt-1.6.2.tar.bz2"
+    url      = "https://gnupg.org/ftp/gcrypt/libgcrypt/libgcrypt-1.7.6.tar.bz2"
 
-    version('1.6.2', 'b54395a93cb1e57619943c082da09d5f')
+    version('1.7.6', '54e180679a7ae4d090f8689ca32b654c')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libgcrypt/package.py]: Can you keep 1.6.2?

Comment (gartung) [inline var/spack/repos/builtin/packages/libgcrypt/package.py]: OK

File: var/spack/repos/builtin/packages/libgpg-error/package.py [hunk 3]
```diff
@@ -32,7 +32,6 @@ class LibgpgError(AutotoolsPackage):
        SmartCard Daemon and possibly more in the future. """
 
     homepage = "https://www.gnupg.org/related_software/libgpg-error"
-    url      = "http://gd.tuwien.ac.at/pub/gnupg/libgpg-error/libgpg-error-1.18.tar.bz2"
+    url      = "https://gnupg.org/ftp/gcrypt/libgpg-error/libgpg-error-1.27.tar.bz2"
 
-    version('1.21', 'ab0b5aba6d0a185b41d07bda804fd8b2')
-    version('1.18', '12312802d2065774b787cbfc22cc04e9')
+    version('1.27', '5217ef3e76a7275a2a3b569a12ddc989')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libgpg-error/package.py]: Can you keep the old versions?

File: var/spack/repos/builtin/packages/npth/package.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    homepage = "https://gnupg.org/software/npth/index.html"
+    url      = "https://gnupg.org/ftp/gcrypt/npth/npth-1.4.tar.bz2"
+
+    version('1.4', '76cef5542e0db6a339cf960641ed86f8')
+
+
+    def configure_args(self):
+        args = []
+        return args
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/npth/package.py]: Delete this function if not used.

File: var/spack/repos/builtin/packages/gnupg/package.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    depends_on('libgpg-error')
+    depends_on('npth')
+
+    def configure_args(self):
+        args = []
+        args += ['--with-npth-prefix=%s'%self.spec['npth'].prefix]
+        args += ['--with-libgcrypt-prefix=%s'%self.spec['libgcrypt'].prefix]
+        args += ['--with-libksba-prefixx=%s'%self.spec['libksba'].prefix]
+        args += ['--with-libassuan-prefix=%s'%self.spec['libassuan'].prefix]
+        args += ['--with-libpgp-error-prefix=%s'%self.spec['libgpg-error'].prefix]
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gnupg/package.py]: Why continuously append to the list when you can create a single list and return it?

Comment (gartung) [inline var/spack/repos/builtin/packages/gnupg/package.py]: Because

**Comment by adamjstewart** [comment]
Not sure if these are from a previous commit or not, but the current flake8 warnings look like:
```
var/spack/repos/builtin/packages/gnupg/package.py:1: [E902] TokenError: EOF in multi-line statement
var/spack/repos/builtin/packages/gnupg/package.py:45: [E126] continuation line over-indented for hanging indent
var/spack/repos/builtin/packages/gnupg/package.py:46: [E999] IndentationError: unexpected indent
var/spack/repos/builtin/packages/gnupg/package.py:46: [E111] indentation is not a multiple of four
var/spack/repos/builtin/packages/gnupg/package.py:46: [E113] unexpected indentation
var/spack/repos/builtin/packages/gnupg/package.py:47: [E111] indentation is not a multiple of four
var/spack/repos/builtin/packages/gnupg/package.py:48: [E111] indentation is not a multiple of four
var/spack/repos/builtin/packages/gnupg/package.py:49: [E111] indentation is not a multiple of four
var/spack/repos/builtin/packages/gnupg/package.py:49: [E501] line too long (80 > 79 characters)
var/spack/repos/builtin/packages/libassuan/package.py:29: [W291] trailing whitespace
var/spack/repos/builtin/packages/libksba/package.py:29: [W291] trailing whitespace
var/spack/repos/builtin/packages/npth/package.py:29: [W291] trailing whitespace
```
The unexpected indentation error is causing all of the unit tests to fail.

File: var/spack/repos/builtin/packages/gnupg/package.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    depends_on('libgpg-error')
+    depends_on('npth')
+
+    def configure_args(self):
+        args = ['--with-npth-prefix=%s' % self.spec['npth'].prefix,
+                '--with-libgcrypt-prefix=%s' % self.spec['libgcrypt'].prefix,
+                '--with-libksba-prefixx=%s' % self.spec['libksba'].prefix,
+                '--with-libassuan-prefix=%s' % self.spec['libassuan'].prefix,
+                '--with-libpgp-error-prefix=%s' %
+                self.spec['libgpg-error'].prefix]
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gnupg/package.py]: If you still see problems, I usually use a format like:
```python
args = [
    'foo',
    'bar'
]
```
You can also run `flake8` locally to identify all of these problems without having to wait for Travis. See [here](http://spack.readthedocs.io/en/latest/contribution_guide.html#flake8-tests) for more info.

Comment (gartung) [inline var/spack/repos/builtin/packages/gnupg/package.py]: I had an extra ] on the npth line.

**Pull request merged**